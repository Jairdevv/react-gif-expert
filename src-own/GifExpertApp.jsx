import { useState } from "react";
import { AddCategory } from "./components/addCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([]);

  const addCategory = (newCategory) => {
    if (categories.includes(newCategory.toLowerCase())) return;
    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory onNewCategory={addCategory} />

      {categories.map((category) => (
        <GifGrid category={category} key={category} />
      ))}
    </>
  );
};
