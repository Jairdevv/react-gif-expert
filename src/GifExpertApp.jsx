import { useState } from "react";
import { GifGrid, AddCategory } from "./components/";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["One punch"]);

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory.toLowerCase())) return;
    // console.log(newCategory);
    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1>GifExpertApp</h1>

      <AddCategory
        onNewCategory={(value) => onAddCategory(value)}
        // setCategories={setCategories}
      />

      {/* <button onClick={onAddCategory}>Agregar</button> */}
      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
