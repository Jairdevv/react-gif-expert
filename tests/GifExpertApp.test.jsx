import { render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src-own/GifExpertApp";

describe("Pruebas en <GifExpertApp/>", () => {
  test("debe de hacer match con el snapshot", () => {
    const { container } = render(<GifExpertApp />);
    // screen.debug();
    expect(container).toMatchSnapshot();
  });

  test("debe de tener 0 categorias", () => {
    render(<GifExpertApp />);
    // screen.debug();

    expect(screen.queryByText("GifGrid")).toBeNull();
  });

  test("should ", () => {});
});
