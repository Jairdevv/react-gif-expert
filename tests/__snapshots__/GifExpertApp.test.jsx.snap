// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <h3>
    GifExpertApp
  </h3>
  <form
    action=""
    aria-label="form"
  >
    <input
      type="text"
      value=""
    />
  </form>
</div>
`;
